### PR TITLE
MLDBFB-509 unprintable char issue test

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: testing/MLDBFB-509_pushed_non_printable_char_cant_query.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class PushNonPrintableCharCantReadTest(MldbUnitTest):  # noqa
+
+    @unittest.expectedFailure
+    def test_unprintable_row_name(self):
+        ds = mldb.create_dataset({'id' : 'ds1', 'type' : 'sparse.mutable'})
+        barbarous_name = 'coco' + chr(17)
+        ds.record_row(barbarous_name, [['colA', 1, 0]])
+        ds.commit()
+
+        # crash!
```
Comment (guyd) [inline testing/MLDBFB-509_pushed_non_printable_char_cant_query.py]: If MLDB crashes would that make the unit test to fail?  I think so and therefore we should skip the test not marked them as `expectedFailure`.

Comment (FinchPowers) [inline testing/MLDBFB-509_pushed_non_printable_char_cant_query.py]: C'est un mauvais étiquette. MLDB ne crash pas mais retourne un 400. D'ailleurs les tests ont passé.

**Comment by guyd** [comment]
+1

**Comment by nicolaskruchten** [comment]
please, please let's use "crash" to mean "MLDB exits with non-zero error
code" and not "4xx or 5xx errors returned over HTTP"

On Mon, May 2, 2016 at 5:12 PM, Guy Dumais <notifications@github.com> wrote:

> +1
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mldbai/mldb/pull/346#issuecomment-216365784>
>



-- 
Nicolas Kruchten, Eng  |  Staff Software Engineer at Datacratic  |
http://datacratic.com/

**Pull request merged**